src/runtime/interop.cs

```diff
@@ -326,37 +343,57 @@ internal class TypeOffset {
         public static int nb_true_divide = 0;
         public static int nb_inplace_floor_divide = 0;
         public static int nb_inplace_true_divide = 0;
-#if (PYTHON25 || PYTHON26 || PYTHON27 || PYTHON32 || PYTHON33 || PYTHON34)
+#if (PYTHON25 || PYTHON26 || PYTHON27 || PYTHON32 || PYTHON33 || PYTHON34 || PYTHON35)
         /* Added in release 2.5 */
         public static int nb_index = 0;
 #endif
+#if (PYTHON35)
+        public static int nb_matrix_multiply;
+        public static int nb_inplace_matrix_multiply;
+#endif
         //} PyNumberMethods;
 //typedef struct {
-        public static int mp_length = 0;
-        public static int mp_subscript = 0;
-        public static int mp_ass_subscript = 0;
-//} PyMappingMethods;
-//typedef struct {
         public static int sq_length = 0;
         public static int sq_concat = 0;
         public static int sq_repeat = 0;
         public static int sq_item = 0;
+#if (PYTHON35)
+        public static int was_sq_slice = 0;
+#else
         public static int sq_slice = 0;
+#endif
         public static int sq_ass_item = 0;
+#if (PYTHON35)
+        public static int was_sq_ass_slice = 0;
+#else
         public static int sq_ass_slice = 0;
+#endif
         public static int sq_contains = 0;
         /* Added in release 2.0 */
         public static int sq_inplace_concat = 0;
         public static int sq_inplace_repeat = 0;
-//} PySequenceMethods;
```

Comment:
@tonyroberts now I agree with you. Note that I did not make this change, most likely this is from @sdpython, but it slipped through my review. this is quite surprising that the order is switched intentionally in cpython due to conflicting situations, according to the comment in cpython source code:

https://github.com/python/cpython/blob/3e91c558fed0e803369f8d2324f102a042cf04e3/Include/object.h#L465